Code/Framework/AzFramework/AzFramework/Entity/EntityOwnershipService.h

```diff
@@ -92,6 +92,13 @@ namespace AzFramework
         virtual void SetEntitiesRemovedCallback(OnEntitiesRemovedCallback onEntitiesRemovedCallback) = 0;
         virtual void SetValidateEntitiesCallback(ValidateEntitiesCallback validateEntitiesCallback) = 0;
 
+        /**
+         * Called when an entity is in the process of being destroyed
+         *
+         * \param entityId The entity Id of the entity being destroyed.
+         */
+        virtual void HandleEntityBeingDestroyed(const AZ::EntityId& entityId) = 0;
```

Comment:
nit: This is super minor and we don't do this every where already, but as `EntityId` is just a type safe wrapper around a `u64`, we can pass by value and not `const` ref